kratos/includes/linear_master_slave_constraint.h

```diff
@@ -387,6 +387,28 @@ class LinearMasterSlaveConstraint
         return mMasterDofsVector;
     }
 
+    /**
+     * @brief This methgod directly applies the master/slave relationship
+     * @param rCurrentProcessInfo the current process info instance
+     */
+    void Apply(const ProcessInfo& rCurrentProcessInfo) override
+    {
+        // Creating the vectors
+        Vector slave_dofs_values(mSlaveDofsVector.size());
```

Comment:
I would suggest to avoid the allocation of the slave_dofs_values vector.

this could be done by doing

~~~cpp
Vector master_dofs_values(mMasterDofsVector.size());

        for (IndexType i = 0; i < mMasterDofsVector.size(); ++i) {
            master_dofs_values[i] = mMasterDofsVector[i]->GetSolutionStepValue();
        }

        for (IndexType i = 0; i < mSlaveDofsVector.size(); ++i) {
            double aux = mConstantVector[i];
            for(IndexType j=0; j<mRelationMatrix.size2(); ++j)
                   aux += mRelationmatrix(i,j)*slave_dofs_values[j];
                 
            #pragma omp atomic
            mSlaveDofsVector[i]->GetSolutionStepValue() += aux;
        }
~~~

Reply:
It makes sense